### PR TITLE
Hide classes into impl package

### PR DESCRIPTION
**Pull request by moarychan**
# Description

- Move classes into the implementation package.
- Rename `AbstractAzureMessageConverter` to `AbstractJacksonAzureMessageConverter`.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/spring/spring-messaging-azure-eventhubs/src/main/java/com/azure/spring/messaging/eventhubs/implementation/core/listener/adapter/BatchMessagingMessageListenerAdapter.java [hunk 1]
```diff
@@ -48,7 +48,7 @@ public void onMessage(EventBatchContext eventBatchContext) {
 
     @Override
     @SuppressWarnings("unchecked")
-    public AbstractAzureMessageConverter<EventBatchContext, EventData> getMessageConverter() {
-        return (AbstractAzureMessageConverter<EventBatchContext, EventData>) super.getMessageConverter();
+    public AbstractJacksonAzureMessageConverter<EventBatchContext, EventData> getMessageConverter() {
```
Comment (saragluna) [inline sdk/spring/spring-messaging-azure-eventhubs/src/main/java/com/azure/spring/messaging/eventhubs/implementation/core/listener/adapter/BatchMessagingMessageListenerAdapter.java:51]: Does this have to be an `AbstractJacksonAzureMessageConverter`?

File: sdk/spring/spring-messaging-azure-eventhubs/src/main/java/com/azure/spring/messaging/eventhubs/implementation/core/listener/adapter/RecordMessagingMessageListenerAdapter.java [hunk 2]
```diff
@@ -49,8 +49,8 @@ public void onMessage(EventContext eventContext) {
 
     @Override
     @SuppressWarnings("unchecked")
-    public AbstractAzureMessageConverter<EventData, EventData> getMessageConverter() {
-        return (AbstractAzureMessageConverter<EventData, EventData>) super.getMessageConverter();
+    public AbstractJacksonAzureMessageConverter<EventData, EventData> getMessageConverter() {
```
Comment (saragluna) [inline sdk/spring/spring-messaging-azure-eventhubs/src/main/java/com/azure/spring/messaging/eventhubs/implementation/core/listener/adapter/RecordMessagingMessageListenerAdapter.java:52]: same

File: sdk/spring/spring-messaging-azure-servicebus/src/main/java/com/azure/spring/messaging/servicebus/implementation/core/listener/adapter/RecordMessagingMessageListenerAdapter.java [hunk 3]
```diff
@@ -42,7 +42,7 @@ public void onMessage(ServiceBusReceivedMessageContext messageContext) {
 
     @Override
     @SuppressWarnings("unchecked")
-    public AbstractAzureMessageConverter<ServiceBusReceivedMessage, ServiceBusMessage> getMessageConverter() {
-        return (AbstractAzureMessageConverter<ServiceBusReceivedMessage, ServiceBusMessage>) super.getMessageConverter();
+    public AbstractJacksonAzureMessageConverter<ServiceBusReceivedMessage, ServiceBusMessage> getMessageConverter() {
```
Comment (saragluna) [inline sdk/spring/spring-messaging-azure-servicebus/src/main/java/com/azure/spring/messaging/servicebus/implementation/core/listener/adapter/RecordMessagingMessageListenerAdapter.java:45]: same

**Review by saragluna — CHANGES_REQUESTED** [review]

File: sdk/spring/spring-messaging-azure-eventhubs/src/main/java/com/azure/spring/messaging/eventhubs/implementation/core/listener/adapter/BatchMessagingMessageListenerAdapter.java (hunk 1 above)
Comment (moarychan) [reply sdk/spring/spring-messaging-azure-eventhubs/src/main/java/com/azure/spring/messaging/eventhubs/implementation/core/listener/adapter/BatchMessagingMessageListenerAdapter.java]: Updated to the interface `AzureMessageConverter`.

**Review by saragluna — APPROVED** [review]

File: sdk/spring/spring-integration-azure-eventhubs/src/test/java/com/azure/spring/integration/eventhubs/inbound/EventHubsInboundChannelAdapterTests.java [hunk 4]
```diff
@@ -116,15 +116,15 @@ void setInstrumentationId() {
 
     @Test
     void setMessageConverter() {
-        AbstractAzureMessageConverter<EventData, EventData> converter = mock(EventHubsMessageConverter.class);
+        AbstractJacksonAzureMessageConverter<EventData, EventData> converter = mock(EventHubsMessageConverter.class);
```
Comment (saragluna) [inline sdk/spring/spring-integration-azure-eventhubs/src/test/java/com/azure/spring/integration/eventhubs/inbound/EventHubsInboundChannelAdapterTests.java:119]: Should be AzureMessageConverter

File: sdk/spring/spring-integration-azure-eventhubs/src/test/java/com/azure/spring/integration/eventhubs/inbound/EventHubsInboundChannelAdapterTests.java [hunk 5]
```diff
@@ -116,15 +116,15 @@ void setInstrumentationId() {
...
         this.adapter.setMessageConverter(converter);
         assertThat(this.adapter).extracting("recordListener").extracting("messageConverter").isEqualTo(converter);
         assertThat(this.adapter).extracting("batchListener").extracting("messageConverter").isNotEqualTo(converter);
     }
 
     @Test
     void setBatchMessageConverter() {
-        AbstractAzureMessageConverter<EventBatchContext, EventData> converter = mock(EventHubsBatchMessageConverter.class);
+        AbstractJacksonAzureMessageConverter<EventBatchContext, EventData> converter = mock(EventHubsBatchMessageConverter.class);
```
Comment (saragluna) [inline sdk/spring/spring-integration-azure-eventhubs/src/test/java/com/azure/spring/integration/eventhubs/inbound/EventHubsInboundChannelAdapterTests.java:127]: same

File: sdk/spring/spring-integration-azure-servicebus/src/test/java/com/azure/spring/integration/servicebus/inbound/ServiceBusInboundChannelAdapterTests.java [hunk 6]
```diff
@@ -103,7 +103,7 @@ void setInstrumentationId() {
 
     @Test
     void setMessageConverter() {
-        AbstractAzureMessageConverter<ServiceBusReceivedMessage, ServiceBusMessage> converter = mock(ServiceBusMessageConverter.class);
+        AbstractJacksonAzureMessageConverter<ServiceBusReceivedMessage, ServiceBusMessage> converter = mock(ServiceBusMessageConverter.class);
```
Comment (saragluna) [inline sdk/spring/spring-integration-azure-servicebus/src/test/java/com/azure/spring/integration/servicebus/inbound/ServiceBusInboundChannelAdapterTests.java:106]: same

File: sdk/spring/spring-integration-azure-eventhubs/src/test/java/com/azure/spring/integration/eventhubs/inbound/EventHubsInboundChannelAdapterTests.java (hunk 5 above)
Comment (moarychan) [reply sdk/spring/spring-integration-azure-eventhubs/src/test/java/com/azure/spring/integration/eventhubs/inbound/EventHubsInboundChannelAdapterTests.java:127]: Fixed in https://github.com/Azure/azure-sdk-for-java/pull/32721

**Pull request merged**